daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Converter.scala

```diff
@@ -65,7 +65,7 @@ object ScriptIds {
 }
 
 final case class AnyTemplate(ty: Identifier, arg: SValue)
-final case class AnyChoice(typeId: Identifier, name: ChoiceName, arg: SValue)
+final case class AnyChoice(ifaceId: Option[Identifier], name: ChoiceName, arg: SValue)
```

Comment:
just wondering if an ADT would make sense here?

```suggestion
abstract class AnyChoice
final case class AnyTemplateChoice(name: ChoiceName, arg: SValue) extends AnyChoice
final case class AnyInterfaceChoice(ifaceId: Identifier, name: ChoiceName, arg: SValue) extends AnyChoice
```

Reply:
:+1: